Strassen Matrix Multiplication/Matrix.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  Matrix.swift
+//  
+//
+//  Created by Richard Ash on 10/28/16.
+//
+//
+
+import Foundation
+
+enum RowOrColumn {
```

Comment:
Yea definitely. I tried this initially but kept getting the error "Type 'RowOrColumn' cannot be nested in generic type 'Matrix'". Do you know of a workaround?